### ISSUE TITLE
📎 Expanding embedded language support by handling embedded content in `JsTemplateExpression`

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#7802]
### Description

This syntax is used by `styled-components` and `emotion` to allow users to define css in their js:

```js
css`
    display: flex;
    color: red;
`;
```

Now that we've figured out embedded parsing, we should be able to easily support this simple use case. 

Where it gets more complicated is when interpolations get involved [as seen here](https://styled-components.com/docs/advanced#tagged-template-literals).

```js
const color = "red";
css`
    display: flex;
    color: ${color};
`;
```

- You can see usages in the wild that use this (or abuse it, really, imo) to do conditional styling.
  - https://sourcegraph.com/search?q=context:global+css%60%5B%5E%60%5D*%5C%24%5C%7B%5B%5E%7D%5D%2B%5C%7D%5B%5E%60%5D*%60+lang:typescript&patternType=regexp&sm=0
- It also doesn't help that official documentation encourages it
  - https://styled-components.com/docs/advanced#theming
- Seems to not be a big problem for graphql (which has it's own interpolation/parameterization like sql does):
  - https://sourcegraph.com/search?q=context:global+gql%60%5B%5E%60%5D*%5C%24%5C%7B%5B%5E%7D%5D%2B%5C%7D%5B%5E%60%5D*%60+lang:typescript&patternType=regexp&sm=0
  - https://sourcegraph.com/search?q=context:global+graphql%60%5B%5E%60%5D*%5C%24%5C%7B%5B%5E%7D%5D%2B%5C%7D%5B%5E%60%5D*%60+lang:typescript&patternType=regexp&sm=0

Here's what we can do about it:
- We could treat interpolations like grit metavariables. It would mean that we wouldn't be _as_ flexible as a template string, but maybe that's a good thing.
- Ignore/warn users when we encounter a `JsTemplateExpression` that contains more than 1 `JsTemplateChunkElement`

**Comment by eMerzh** [comment]
nice to see those coming ❤️ 

indeed interpolations can go a bit crazy, between the simple "string interpolation you have there and the function one where you access the `theme`object  (or other props like `as` or `forwardedAs`) 
like 
```
border: 2px dashed ${({ theme,  }) => theme.colors.primary};
```

and of course can be nested one in another 

```
	position: relative;
	&::before {
		position: absolute;
		left: 0;
		right: 0;
		top: 0;
		bottom: 0;
		content: '';
		z-index: -1;
		${
			props.$highlighted &&
			css`
			z-index: 1;
			background-color: rgba(${hexToRgb(theme.colors.dropzoneHighlighted)}, ${props.$label ? 1 : 0.1});
			border: 2px dashed ${({ theme }) => theme.colors.primary};
			border-radius: 2px;
		`
		};
}
```


indeed the graphql`` support would probablty be easier 🤔

**Comment by talovsky** [comment]
> and of course can be nested one in another

Maybe it's possible to narrow the focus down? E.g. ignore nested expressions? Even a basic support here would be a very welcome addition.

**Comment by ematipico** [comment]
Problem is, even one interpolation will trigger a parsing error because they don't exist in CSS, so maybe we should treat it as grit metavariable

**Comment by siketyan** [comment]
One difficulty here is that the embedded CSS snippets have an implicit declaration block outside.

For example:

```js
const Foo = styled.div`
  color: red;
`;
```

... is equivalent to the following in CSS ...

```css
div.Foo {
  color: red;
}
```

Our CSS parser needs the declarations to be placed inside a declaration block, so we need to use the parser with a dummy block and translate offsets after parsing.

**Comment by ematipico** [comment]
>  so we need to use the parser with a dummy block and translate offsets after parsing

Argh, that's rough. This might mess up the offsets for diagnostics (parser, linting) and rule code actions too.
